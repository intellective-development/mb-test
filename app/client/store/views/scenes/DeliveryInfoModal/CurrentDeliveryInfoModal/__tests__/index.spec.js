/* eslint import/first: 0 */

jest.mock('client/shared/components/higher_order/make_provided'); // avoid importing the full store
jest.mock('legacy_store/models/Cart');

import * as React from 'react';

import { CurrentDeliveryInfoModal } from '../index';

describe('CurrentDeliveryInfoModal', () => {
  it('renders the current_delivery_info view', () => {
    expect(shallow(
      <CurrentDeliveryInfoModal />
    )).toMatchSnapshot();
  });

  it('renders the supplier switching view after routing to it', () => {
    const component = shallow(<CurrentDeliveryInfoModal />);

    // switch to the supplier switching view
    component.instance().deliveryInfoRouteTo('supplier_switching', {supplier_id: 1});
    component.update();

    expect(component).toMatchSnapshot();
  });

  it('renders the waitlist view after routing to it', () => {
    const component = shallow(<CurrentDeliveryInfoModal />);

    // switch to the waitlist view
    component.instance().deliveryInfoRouteTo('address_waitlist', {supplier_id: 1});
    component.update();

    expect(component).toMatchSnapshot();
  });

  it('renders nothing when hidden', () => {
    expect(shallow(
      <CurrentDeliveryInfoModal is_hidden />
    )).toMatchSnapshot();
  });
});
