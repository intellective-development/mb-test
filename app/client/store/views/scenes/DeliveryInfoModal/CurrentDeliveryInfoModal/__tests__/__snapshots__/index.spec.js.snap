// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentDeliveryInfoModal renders nothing when hidden 1`] = `
<Modal
  show={false}
  size="large"
>
  <[object Object]
    deliveryInfoRouteTo={[Function]}
  />
</Modal>
`;

exports[`CurrentDeliveryInfoModal renders the current_delivery_info view 1`] = `
<Modal
  show={true}
  size="large"
>
  <[object Object]
    deliveryInfoRouteTo={[Function]}
  />
</Modal>
`;

exports[`CurrentDeliveryInfoModal renders the supplier switching view after routing to it 1`] = `
<Modal
  show={true}
  size="large"
>
  <[object Object]
    routeBack={[Function]}
    supplier_id={1}
  />
</Modal>
`;

exports[`CurrentDeliveryInfoModal renders the waitlist view after routing to it 1`] = `
<Modal
  show={true}
  size="large"
>
  <[object Object]
    routeBack={[Function]}
  />
</Modal>
`;
